Q1. Write a program using the Regular Exception and create a function that accepts a string and searches it for a valid phone number.
Return the phone number if found.
A valid phone number may be one of the following:
(xxx)-xxx-xxxx
xxx-xxx-xxxx


In [3]:
import re

def find_phone_number(text):
    pattern = re.compile(r'(\(\d{3}\)-\d{3}-\d{4}|\d{3}-\d{3}-\d{4})')
    match = pattern.search(text)
    if match:
        return match.group()
    else:
        return None

try:
    input_text = input("Enter a text to search for a phone number: ")
    phone_number = find_phone_number(input_text)
    
    if phone_number:
        print(f"Valid phone number found: {phone_number}")
    else:
        print("No valid phone number found in the text.")

except Exception as e:
    print(f"An error occurred: {e}")

finally:
    print("Program execution completed.")


Valid phone number found: 555-123-4567
Program execution completed.


Q2. Write a function that employs regular expressions to ensure the password given to the function is strong.
A strong password is defined as follows:
·       at least eight characters long
·       contains one uppercase character
·       contains one lowercase character
·       has at least one digit
·       has at least one special character
[For instance: Christ@123]

In [5]:
import re

def is_strong_password(password):
    # Define the regular expression pattern for a strong password
    pattern = re.compile(r'^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*[@#$%^&+=!]).{8,}$')
    
    # Check if the password matches the pattern
    if pattern.match(password):
        return True
    else:
        return False

# Test the function
password = input("Enter a password: ")

if is_strong_password(password):
    print("Password is strong.")
else:
    print("Password is not strong.")


Password is strong.
